From c68fd3c94c6debdbf11020940c5a6aaee8bc230d Mon Sep 17 00:00:00 2001
From: Feichtmeier <frederik.feichtmeier@gmail.com>
Date: Fri, 15 Mar 2019 14:41:55 +0100
Subject: [PATCH] theme: Update window preview style

 - simplify the close button to use blue, lighter blue and darker blue
   solid disks for normal, hover and active states

 - use a milky, transparent white border for the hover effect of the border

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/461
---
 data/theme/gnome-shell-sass/_common.scss | 29 ++++++++++++------------
 1 file changed, 14 insertions(+), 15 deletions(-)

diff --git a/data/theme/gnome-shell-sass/_common.scss b/data/theme/gnome-shell-sass/_common.scss
index 9e0751c8c..8bf368f6e 100644
--- a/data/theme/gnome-shell-sass/_common.scss
+++ b/data/theme/gnome-shell-sass/_common.scss
@@ -1164,25 +1164,23 @@ StScrollBar {
 //close buttons
 
 .window-close {
-  background-color: white;
+  background-color: $selected_bg_color;
+  color: white;
   border-radius: 24px;
-  border: 4px solid $selected_bg_color;
-  box-shadow: inset 0 -4px 0 0 transparentize($selected_bg_color, 0.5);
-  color: $selected_bg_color;
+  border: 2px solid $selected_bg_color;
   height: 24px;
   width: 24px;
-  -shell-close-overlap: 14px;
+  -shell-close-overlap: 11px;
+  box-shadow: -1px 1px 5px 0px transparentize(black, 0.5);
 
   &:hover {
-    background-color: $selected_bg_color;
-    border-color: white;
-    color: white;
+    background-color: lighten($selected_bg_color, 5%);
+    border-color: lighten($selected_bg_color, 5%);
   }
 
   &:active {
-    background-color: mix(white, $selected_bg_color, 75%);
-    border-color: $selected_bg_color;
-    color: $selected_bg_color;
+    background-color: darken($selected_bg_color, 5%);
+    border-color: darken($selected_bg_color, 5%);
   }
 }
 
@@ -1247,13 +1245,14 @@ StScrollBar {
   }
 
   .window-clone-border {
-    border: 4px solid $selected_bg_color;
-    border-radius: 4px;
+    $_bg: transparentize(white, 0.65);
+    border: 5px solid $_bg;
+    border-radius: 6px;
     // For window decorations with round corners we can't match
     // the exact shape when the window is scaled. So apply a shadow
     // to fix that case
-    box-shadow: inset 0px 0px 0px 1px $selected_bg_color;
-  }
+    box-shadow: inset 0 0 0 1px $_bg;
+}
   .window-caption {
     spacing: 25px;
     color: $selected_fg_color;
-- 
2.31.1

